crates/static-analysis-kernel/src/model/ruleset.rs

```diff
@@ -1,9 +1,132 @@
 use crate::model::rule::Rule;
-use serde::{Deserialize, Serialize};
+use serde::{Deserialize, Deserializer, Serialize};
 
-#[derive(Clone, Deserialize, Debug, Serialize)]
+#[derive(Clone, Debug, Serialize)]
 pub struct RuleSet {
     pub name: String,
     pub description: Option<String>,
-    pub rules: Vec<Rule>,
+    rules: Vec<Rule>,
+    /// A private field to prevent this struct from being created manually. Use [`RuleSet::new`] instead.
+    _private: std::marker::PhantomData<()>,
+}
+
+impl RuleSet {
+    /// Creates a new `RuleSet` from the provided rules, ensuring that the name of each rule
+    /// has the ruleset id prepended in the format `ruleset_id/rule_name`.
+    pub fn new(
+        ruleset_id: impl Into<String>,
```

Comment:
nit(non-blocking): References to this method pass in either a variable called `ruleset_name` or a name field of a ruleset called `ruleset.name`, and it's also used to populate the name field of this struct. For consistency, we could call this `ruleset_name` instead, and update relevant comments to use "ruleset name" instead of "ruleset id".